packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -217,7 +217,7 @@ export interface EcrProps {
    */
   readonly repository: ecr.IRepository;
   /**
-   * Image tag.
+   * Image tag or digest.
    * @default - 'latest'
    */
   readonly tag?: string;
```

Comment:
can we deprecate `tag` and create `tagOrDigest` instead, to better reflect what this prop represents?

Reply:
Sounds good; before I spend the time tweaking things, just a few clarifying questions:

1. based on other parts of CDK, I'm assuming I can just add the new property and insert ``@deprecated use `tagOrDigest` `` (or something) into the docs of this one. That is, I don't need to add a warning to the code or something?
2. I'm also assuming it's fine to do something like `this.props.repository.repositoryUriForTagOrDigest(this.props.tagOrDigest || this.props.tag || 'latest')`, rather than keep the existing behaviour for users of `tag`, but let me know if not

(I'll do a similar deprecation to `EcrImageCodeProps.tag` too, and change all parameter names for the plain functions.)

Reply:
1. Yes, this is fine. No need to add a printed warning message.

2. I'm not sure what you mean by "rather than keep the existing behavior for users of `tag`." Your proposed change doesn't change the behavior for existing users of `tag`, it only prefers the `tagOrDigest` to `tag`. If you like, you can throw an error if both properties are specified, but that's optional. The example you specified is plenty.

Reply:
1. Cool, will do
2. Ah, sorry, I was unclear, I guess I was thinking if someone has tagged an image with a tag like `sha256:my-custom-tag` and thus is relying on getting `url/repo:sha256:my-custom-tag` out of the this system, but, looking into it now, I think that's not a valid tag so this situation is impossible. 

   > A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters. &mdash; https://docs.docker.com/engine/reference/commandline/tag

Reply:
I've done the switch to `tagOrDigest` everywhere.